### PR TITLE
Remove BLangCompilerException with a warning for invalid settings

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/28764

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by hevayo** [comment]
Can we add the settings.toml path to the warning so that users can find it easily.

**Comment by pramodya1994** [comment]
> Can we add the settings.toml path to the warning so that users can find it easily.

What do you think about following warning?
```
warning: invalid 'Settings.toml' file at:/Users/pramodya/.ballerina/Settings.toml due to:Key is not followed by an equals sign on line 4: 353535

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/SettingsProcessor.java [hunk 1]
```diff
@@ -41,23 +43,13 @@
...
-    private static Settings getSettings(InputStream inputStream) {
         Toml toml;
-        try {
-            toml = new Toml().read(inputStream);
+        try (InputStream settingsInputStream = new FileInputStream(settingsPath.toFile())) {
+            toml = new Toml().read(settingsInputStream);
+            return toml.to(Settings.class);
         } catch (IllegalStateException e) {
-            throw new BLangCompilerException("invalid Settings.toml due to " + e.getMessage());
+            out.println("warning: invalid 'Settings.toml' file at:" + settingsPath + " due to:" + e.getMessage());
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/SettingsProcessor.java:51]: This output will break LS if invoked from Project API.

**Review by hevayo — CHANGES_REQUESTED** [review]
Please see inline

**Comment by pramodya1994** [comment]
We can ignore the toml parsing errors and continue the build, the Only exception is when the user has defined a proxy in the `Settings.toml`, and if there's a syntax error, Then we won't consider defined proxy when we set the remote repo.

@hevayo WDYT about this?

**Review by hevayo — CHANGES_REQUESTED** [review]
Can we add warning in a build command for invalid settings files

**Review by hevayo — APPROVED** [review]

**Review by hevayo — APPROVED** [review]

**Pull request merged**